src/Workleap.Extensions.OpenAPI.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -136,15 +136,23 @@ public void ValidateEndpointResponseWithAnnotationType(SymbolAnalysisContext con
         private void ValidateAnnotationWithTypedResult(SymbolAnalysisContext context, AttributeData attribute,
             Dictionary<int, List<ITypeSymbol>> methodSignatureStatusCodeToTypeMap)
         {
-            if (attribute.AttributeClass == null || attribute.ConstructorArguments.Length == 0)
+            if (attribute.AttributeClass == null || attribute.ConstructorArguments.Length == 0 && attribute.NamedArguments.Length == 0)
             {
                 return;
             }
 
-            // For the annotations [ProducesResponseType(<StatusCode>)] and [ProducesResponseType(<typeof()>, <StatusCode>)] 
+            // For the annotations [ProducesResponseType(<StatusCode>)], [ProducesResponseType(<typeof()>, <StatusCode>)] and [ProducesResponseType(<StatusCode>, Type = <typeof()>)] 
             if (attribute.AttributeClass.Equals(this.ProducesResponseSymbol, SymbolEqualityComparer.Default))
             {
-                if (attribute.ConstructorArguments.Length == 1)
+                if (TryGetTypeFromNamedArguments(attribute, out var typeFromNamedArgument))
+                {
+                    if (attribute.ConstructorArguments[0].Value is int statusCodeValue && typeFromNamedArgument != null)
```

Comment:
Should we check `attribute.ConstructorArguments.Length == 1` at the top the condition since we are already checking if the other condition (l.155) and here are are accesing `(attribute.ConstructorArguments[0].Value `without checking it's lenght.

Could be something:

if (attribute.AttributeClass.Equals(this.ProducesResponseSymbol, SymbolEqualityComparer.Default) && attribute.ConstructorArguments.Length == 1 && attribute.ConstructorArguments[0].Value is int statusCodeValue)
{
...
}

Maybe this could simplify a couple of tihngs

Reply:
Good point, decided to split into separate methods and added comments